New object structure 2024-11-27

EstevaoCalera opened the pull request:
The 2024-11-27 update brings some structure changes:

* 'ShowBackground = true' on every annotation

* 'GrenadeType' on main annotations to specify which grenade it refers to

ReneRebsdorf commented:
Awesome, thank you!

You were aster than me, just put it on issuetracker before going to bed
Will review and merge this later today.

ReneRebsdorf commented on de_ancient.txt:
```diff
@@ -84,13 +89,15 @@
 			FontSize = 75
 			FadeInDist = 50.0
 			FadeOutDist = -1.0
+			ShowBackground = true
```
I need to check whether the background is hidden when no text is present. If it needs to be false in that case I will write a test for it

EstevaoCalera replied:
When there is no text, the background doesn't show either. In the following image, the "standing instructions" are empty and nothing is shown, even though ShowBackground = true

![image](https://github.com/user-attachments/assets/13e8b48a-efde-4acf-afd6-5d7588b47b50)

ReneRebsdorf reviewed (changes requested):
Looks good, one comment to verify i game, if you already tested this we can resolve and merge

EstevaoCalera commented:
I have added two commits to fix some errors:

* Removed unnecessary GrenadeType = "smoke" from text annotations

* Changed the file location to fix the issue #22

ReneRebsdorf reviewed (changes requested):
`.github/workflows/main.yml` needs updating as the zip file should be immediately extractable to the csgo folder
so we should just zip the local folder to local.zip for the easiest user experience I think.

This affects :16 and :20

I can do this later today for this PR, and I think we are good to merge then :)

EstevaoCalera commented:
I would love to help you with that workflow, but I don't know exactly how to test this Github Actions. I've never used it before.

How can I test it locally? Is there any VSCode extensions?

ReneRebsdorf commented:
> I would love to help you with that workflow, but I don't know exactly how to test this Github Actions. I've never used it before.
> 
> How can I test it locally? Is there any VSCode extensions?

I have never looked into it, but I've heard https://github.com/nektos/act might be what you are looking for.
But I've mostly just relied on making branches, or even other repos, to test my builds and verify them before merging.

In other words, I just use the actual github workflows to verify their functionality. It would be great to be able to run them locally as I much prefer testing before commit+push.
